cfme/tests/intelligence/reports/test_validate_chargeback_report.py

```diff
@@ -0,0 +1,493 @@
+            "network_used_cost": network_used_cost,
+            "disk_used_cost": disk_used_cost}
+
+
+@pytest.yield_fixture(scope="module")
+def chargeback_report_default(vm_ownership, assign_compute_default_rate, provider):
+    # Create a Chargeback report based on the default Compute rate; Queue the report
+    owner = vm_ownership
+    data = {'menu_name': 'cb_' + provider.name,
+            'title': 'cb_' + provider.name,
+            'base_report_on': 'Chargebacks',
+            'report_fields': ['Memory Used', 'Memory Used Cost', 'Owner',
+            'CPU Used', 'CPU Used Cost',
+            'Disk I/O Used', 'Disk I/O Used Cost',
+            'Network I/O Used', 'Network I/O Used Cost'],
+            'filter_show_costs': 'Owner',
+            'filter_owner': owner,
+            'interval_end': 'Today (partial)'}
+    report = CustomReport(is_candu=True, **data)
+    report.create()
+
+    report.queue(wait_for_finish=True)
+
+    yield list(report.get_saved_reports()[0].data[0].rows)
+    report.delete()
+
+
+@pytest.yield_fixture(scope="module")
+def chargeback_report_custom(vm_ownership, assign_compute_custom_rate, provider):
+    # Create a Chargeback report based on a custom Compute rate; Queue the report
+    owner = vm_ownership
+    data = {'menu_name': 'cb_custom_' + provider.name,
+            'title': 'cb_custom' + provider.name,
+            'base_report_on': 'Chargebacks',
+            'report_fields': ['Memory Used', 'Memory Used Cost', 'Owner',
+            'CPU Used', 'CPU Used Cost',
+            'Disk I/O Used', 'Disk I/O Used Cost',
+            'Network I/O Used', 'Network I/O Used Cost'],
+            'filter_show_costs': 'Owner',
+            'filter_owner': owner,
+            'interval_end': 'Today (partial)'}
+    report = CustomReport(is_candu=True, **data)
+    report.create()
+
+    report.queue(wait_for_finish=True)
+
+    yield list(report.get_saved_reports()[0].data[0].rows)
+    report.delete()
+
+
+@pytest.yield_fixture(scope="module")
+def new_compute_rate():
+    # Create a new Chargeback compute rate
+    try:
+        desc = 'custom_' + fauxfactory.gen_alphanumeric()
+        ccb = cb.ComputeRate(description=desc,
+                         cpu_used=(1.00, cb.DAILY),
+                         disk_io=(1.00, cb.HOURLY),
+                         compute_fixed_1=(0, cb.DAILY),
+                         compute_fixed_2=(0, cb.MONTHLY),
+                         mem_alloc=(0, cb.DAILY),
+                         mem_used=(1.50, cb.DAILY),
+                         net_io=(1.70, cb.HOURLY))
+        ccb.create()
+        yield desc
+    finally:
+        ccb.delete()
+
+
+# Tests to validate costs reported in the Chargeback report for various metrics.
+# The value reported in the Chargeback report should be approximately equal to the
+# costs estimated in the chargeback_costs_default/chargeback_costs_custom fixtures.
+def test_validate_cpu_usage_default_rate(
```

Comment:
def func(chargeback_costs_default, chargeback_report_default, soft_assert):
for groups in chargeback_report_default:
    if groups["CPU Used Cost"]:
        estimated_cpu_usage_cost = chargeback_costs_default['cpu_used_cost']
        cost_from_report = groups["CPU Used Cost"]
        cost = re.sub(r'[$,]', r'', cost_from_report)
        soft_assert(
            estimated_cpu_usage_cost - 1.0 <= float(cost) <=
            estimated_cpu_usage_cost + 1.0)
        break

def test_validate_cpu_usage_default_rate(
        chargeback_costs_default, soft_assert,
        chargeback_report_default):
    func(chargeback_costs_default, chargeback_report_default, soft_assert)